### PR TITLE
Delete messages improvements/fixes

### PR DESCRIPTION
**Pull request by Ivansss**

### CODE REVIEW

File: NextcloudTalk/NCChatMessage.m [hunk 1]
```diff
@@ -141,11 +168,12 @@ - (BOOL)isMessageFromUser:(NSString *)userId
     return [self.actorId isEqualToString:userId] && [self.actorType isEqualToString:@"users"];
 }
 
-- (BOOL)isDeletableForUserId:(NSString *)userId andParticipantType:(NCParticipantType)participantType
+- (BOOL)isDeletableForAccount:(TalkAccount *)account andParticipantType:(NCParticipantType)participantType
 {
     NSInteger sixHoursAgoTimestamp = [[NSDate date] timeIntervalSince1970] - (6 * 3600);
-    if ([self.messageType isEqualToString:kMessageTypeComment] && !self.file && self.timestamp >= sixHoursAgoTimestamp &&
-        (participantType == kNCParticipantTypeOwner || participantType == kNCParticipantTypeModerator || [self isMessageFromUser:userId])) {
+    BOOL canServerDeleteMessages = [[NCSettingsController sharedInstance] serverHasTalkCapability:kCapabilityChatReadStatus forAccountId:account.accountId];
```
Comment (SystemKeeper) [inline NextcloudTalk/NCChatMessage.m:174]: Really Read-Status here ?

Comment (Ivansss) [reply NextcloudTalk/NCChatMessage.m:174]: 😬🙈 Thanks! 😉

Comment (Ivansss) [reply NextcloudTalk/NCChatMessage.m:174]: Fixed in c8623f8a33d1f853b08b2b7dd926a557a22396ca

Comment (SystemKeeper) [reply NextcloudTalk/NCChatMessage.m:174]: 👍

**Pull request merged**